integration-tests/smoke/cron_test.go

```diff
@@ -59,3 +59,72 @@ func TestCronBasic(t *testing.T) {
 		}
 	}, "2m", "3s").Should(gomega.Succeed())
 }
+
+func TestCronJobReplacement(t *testing.T) {
```

Comment:
There are a lot of gomega.Eventually calls in the tests below. This usually means that this test could become flakey at some point. Is there any way to be notified when the end condition has been met rather than polling here?

Reply:
I don't see a good way of notifying on completions, especially something that would not necessitate a large refactor